Mathlib/Data/Fintype/BigOperators.lean

```diff
@@ -296,34 +296,26 @@ theorem Fintype.prod_sum_type (f : Sum α₁ α₂ → M) :
 #align fintype.prod_sum_type Fintype.prod_sum_type
 #align fintype.sum_sum_type Fintype.sum_sum_type
 
-@[to_additive (attr := simp)]
-theorem Fintype.prod_sum_type_right (f : Sum α₁ α₂ → M) :
-    ∏ x, f x = (∏ a₂, f (Sum.inr a₂)) * ∏ a₁, f (Sum.inl a₁) := by
-  rw [mul_comm]
-  exact prod_sum_type _
-
-@[to_additive Fintype.sum_prod_type]
+@[to_additive (attr := simp) Fintype.sum_prod_type ]
 theorem Fintype.prod_prod_type [CommMonoid γ] {f : α₁ × α₂ → γ} :
     ∏ x : α₁ × α₂, f x = ∏ x : α₁, ∏ y : α₂, f (x, y) :=
-  prod_finset_product univ univ (fun _ ↦ univ) <|
-      fun p => ⟨fun _ => ⟨mem_univ p.1, mem_univ p.2⟩, fun _ => mem_univ p⟩
+  Finset.prod_product
 
 /-- An uncurried version of `Finset.prod_prod_type`. -/
-@[to_additive Fintype.sum_prod_type' "An uncurried version of `Finset.sum_prod_type`"]
+@[to_additive (attr := simp) Fintype.sum_prod_type' "An uncurried version of `Finset.sum_prod_type`"]
 theorem Fintype.prod_prod_type' [CommMonoid γ] {f : α₁ → α₂ → γ} :
     ∏ x : α₁ × α₂, f x.1 x.2 = ∏ x : α₁, ∏ y : α₂, f x y :=
-  prod_prod_type
+  Finset.prod_product'
 
 @[to_additive Fintype.sum_prod_type_right]
 theorem Fintype.prod_prod_type_right [CommMonoid γ] {f : α₁ × α₂ → γ} :
-    ∏ x : α₁ × α₂, f x = ∏ y : α₂, ∏ x : α₁, f (x, y) := by
-  rw [prod_comm]
-  exact prod_prod_type
+    ∏ x : α₁ × α₂, f x = ∏ y : α₂, ∏ x : α₁, f (x, y) :=
```

Comment:
```suggestion
    ∏ x, f x = ∏ y, ∏ x, f (x, y) :=
```
and elsewhere

Reply:
done, also elsewhere, expect in some places in the uncurried versions, where it didn't work.